src/python/bot/fuzzers/ml/gradientfuzz/train.py

```diff
@@ -0,0 +1,163 @@
+
+    Args:
+        config (dict): Run configuration dictionary (usually from
+            parser.parse_args()).
+
+    Returns:
+        model (keras.Model): loaded model from config.
+    """
+  latest_filename = tf.train.latest_checkpoint(
+      utils.get_full_path(config['run_name']))
+
+  print('Loading \"{}\" model from {}...'.format(config['architecture'],
+                                                 latest_filename))
+  model = models.make_model_from_layer(
+      constants.ARCHITECTURE_MAP[config['architecture']],
+      config['output_dim'],
+      config['input_shape'],
+      hidden_layer_dim=config['num_hidden'])
+  model.load_weights(latest_filename).expect_partial()
+
+  # Update 'cur_epoch' in config, since training loop doesn't do it.
+  config['cur_epoch'] = int(
+      latest_filename[-constants.CP_EPOCH_NUM_DIGITS + 1:])
+  print('Last saved epoch: {}.'.format(config['cur_epoch']))
+  utils.save_model_config(config)
+
+  # Training options.
+  model.compile(
+      optimizer=constants.OPTIMIZER_MAP[config['optimizer']](
+          learning_rate=config['lr']),
+      loss=keras.losses.BinaryCrossentropy(from_logits=False),
+      metrics=[model_utils.BitmapAcc(),
+               model_utils.NeuzzJaccardAcc()])
+
+  return model
+
+
+def get_new_model_from(config):
+  """
+    Creates and returns a NEW compiled model from arguments.
+    Assumes NOT using a pre-determined config (e.g. --neuzz-config),
+    and assumes the model doesn't already exist.
+
+    Args:
+        config (dict): Run configuration dictionary (usually from
+            parser.parse_args()).
+
+    Returns:
+        model (keras.Model): new model constructed from config.
+    """
+
+  # Model options.
+  model = models.make_model_from_layer(
+      constants.ARCHITECTURE_MAP[config['architecture']],
+      config['output_dim'],
+      config['input_shape'],
+      hidden_layer_dim=config['num_hidden'])
+
+  # Training options.
+  model.compile(
+      optimizer=constants.OPTIMIZER_MAP[config['optimizer']](
+          learning_rate=config['lr']),
+      loss=keras.losses.BinaryCrossentropy(from_logits=False),
+      metrics=[model_utils.BitmapAcc(),
+               model_utils.NeuzzJaccardAcc()])
+
+  return model
+
+
+def train_model(model, config, train_dataset, val_dataset):
+  """
+    Constructs keras.Callbacks to save model progress and runs model.fit().
+
+    Args:
+        model (keras.Model): Model to train.
+        config (dict): Configuration as constructed by `config_from_args()`
+            in `utils.py`.
+        train_dataset (ProgramDataset): Training set for model.
+        val_dataset (ProgramDatset): Val set for model (backprop doesn't
+            happen on these).
+
+    Returns:
+        tf.callbacks.History object documenting training history.
+    """
+  utils.save_model_config(config)
+  callback_list = model_utils.get_callbacks(config)
+  _ = model.fit(
```

Comment:
no need for the `_ = ` part.

Reply:
Right -- I was planning on doing something with the `History` object initially and decided against it.